src/System.Runtime.Extensions/src/System/Environment.WinRT.cs

```diff
@@ -0,0 +1,138 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Windows.Foundation.Metadata;
+using Windows.Storage;
+using System.IO;
+
+namespace System
+{
+    public static partial class Environment
+    {
+        private static string GetFolderPathCore(SpecialFolder folder, SpecialFolderOption option)
+        {
+            // For testing we'll fall back if the needed APIs aren't present.
+            //
+            // We're not honoring the special folder options (noverify/create) for a few reasons. One, most of the
+            // folders always exist (e.g. it is moot). Two, most locations are inaccessible from an appcontainer
+            // currently - making it impossible to answer the question of existence or create if necessary. Thirdly,
+            // the Win32 API would create these folders with very specific ACLs, which even in the cases we can create
+            // are a significant compat risk (trying to replicate internal Windows behavior- it is documented that they
+            // set specific ACLs, but not which ones).
+            if (ApiInformation.IsTypePresent("Windows.Storage.UserDataPaths"))
```

Comment:
I asked this on the original PR, but I'd still be interested in understanding what the cost of this is and what kind of overhead it's adding to GetSpecialFolder calls, whether we should consider caching it or not, etc.  It's of course possible it's inconsequential, but it'd be good to know for sure.